lib/wasix/src/net/socket.rs

```diff
@@ -114,37 +114,40 @@ pub enum WasiSocketOption {
     Proto,
 }
 
-impl From<Sockoption> for WasiSocketOption {
-    fn from(opt: Sockoption) -> Self {
+impl TryFrom<Sockoption> for WasiSocketOption {
+    type Error = Errno;
+
+    fn try_from(opt: Sockoption) -> Result<Self, Self::Error> {
         use WasiSocketOption::*;
         match opt {
-            Sockoption::Noop => Noop,
```

Comment:
If you want, you don't have to use `Ok` on the RHS of `=>` statements, you can do:
```rust
Ok(match opt {...,  _ => return Err(Errno::Inval) })
```

Reply:
Sure!